runtime/src/main/java/org/corfudb/runtime/view/stream/BackpointerStreamView.java

```diff
@@ -290,13 +290,15 @@ protected boolean followBackpointers(final UUID streamId,
 
     }
 
-    protected BackpointerOp resolveCheckpoint(final QueuedStreamContext context, ILogData data) {
+    protected BackpointerOp resolveCheckpoint(final QueuedStreamContext context, ILogData data,
+                                              long maxGlobal) {
         if (data.hasCheckpointMetadata()) {
             CheckpointEntry cpEntry = (CheckpointEntry)
                     data.getPayload(runtime);
 
             if (context.checkpointSuccessId == null &&
-                    cpEntry.getCpType() == CheckpointEntry.CheckpointEntryType.END) {
+                    cpEntry.getCpType() == CheckpointEntry.CheckpointEntryType.END
+             && Long.decode(cpEntry.getDict().get(CheckpointEntry.CheckpointDictKey.START_LOG_ADDRESS)) <= maxGlobal) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getDict"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)